### PR TITLE
Support exclude in Doc.from_docs

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

Support `exclude` in `Doc.from_docs` for `spans`, `tensor`, and `user_data`.

### Types of change

Enhancement.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
See: https://github.com/explosion/spaCy/discussions/10597

**Comment by adrianeboyd** [comment]
@explosion-bot please test_gpu

File: website/docs/api/doc.md [hunk 1]
```diff
@@ -403,6 +404,8 @@ Concatenate multiple `Doc` objects to form a new one. Raises an error if the
 | `docs`              | A list of `Doc` objects. ~~List[Doc]~~                                                                            |
 | `ensure_whitespace` | Insert a space between two adjacent docs whenever the first doc does not end in whitespace. ~~bool~~              |
 | `attrs`             | Optional list of attribute ID ints or attribute name strings. ~~Optional[List[Union[str, int]]]~~                 |
+| _keyword-only_      |                                                                                                                   |
+| `exclude`           | String names of Doc attributes to exclude. Supported: `spans`, `tensor`, `user_data`. ~~Iterable[str]~~           |
```
Comment (svlandeg) [inline website/docs/api/doc.md:408]:
```suggestion
| `exclude` <Tag variant="new">3.3</Tag>           | String names of Doc attributes to exclude. Supported: `spans`, `tensor`, `user_data`. ~~Iterable[str]~~           |
```
+ reformat

**Review by svlandeg — APPROVED** [review]
Looks good to me!

**Pull request merged**